Automattic/jetpack
# Call ensure_options methods before initialization

#### Changes proposed in this Pull Request:

* Make Config call the `ensure_options_$feature` methods before the initialization of the packages

#### Other information:


#### Jetpack product discussion


p1660822904458439-slack-C02NENKA4GN

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






At this point, nothing should change, so test for regressions.

Let's test Sync to see if the behavior was kept the same.

* Actiave only VideoPress plugin
* Go to Jetpack Debug > Sync Data Settings Tester and checks the result
* Activate Protect and check the results again. `active_plugins` and `stylesheet` should have been added to the list of options. 

Run the test above on trunk and with this branch checked out. The results must be the same.

Check the docs to see if they are clear and correct.
